library/FrontRow/App/Database.hs

```diff
@@ -58,7 +64,13 @@ runDB
   -> m a
 runDB action = do
   pool <- asks getSqlPool
-  runSqlPool action pool
+  mTimeout <- asks getStatementTimeout
+
+  flip runSqlPool pool $ do
+    for_ mTimeout $ \timeout ->
+      let timeoutMillis = unSeconds timeout * 1000
+      in [executeQQ| SET statement_timeout = #{timeoutMillis} |]
```

Comment:
:ghost: 

Doing this in `runDB` adds some round trip latency to every DB transaction. You could consider modifying the pool to set `statement_timeout` when a connection is first created, so all transactions use it implicitly and you don't have to pay for the round trip each time you open a transaction. Alternatively, you could find a way to have persistent send `begin; set statement_timeout = ?;` through the connection instead of just `begin;` when the transaction opens, so you don't have to waste a round trip (since you are sending `begin;` anyway).

Probably not a big deal, but I'm just throwing this out there if you start to see transaction execution times for smaller transactions increase.

Reply:
ohai :wave: 

@cbeav this is no different than the implementation in fancy-api right? If the tax David described were to bite us, it would most certainly bite us there.

Reply:
Up-streaming an ability to set this on Pool initialization is definitely the best answer though, IMO. I imagine that's `persistent`-owned stuff.

Reply:
Haha, hey there. This is just a complete copy of our current `fancy-api` implementation. We're paying that tax there as well. However, would be nice to pay in neither!

Check out what I just put out. It no longer needs `getStatementTimeout`, but relies on loaded PG config from env. How do we feel about that? I like that it simplifies just relying on the PG config in one place, but it does make this more of a buried feature of the library.

Reply:
Also no tax!

Reply:
Whoah awesome! I totally forgot we hooked into the pool to do the IAM auth stuff. I thought it would be harder.